src/customParser.ts

```diff
@@ -0,0 +1,330 @@
+    const type = arraysList[arrayId];
+
+    if ((type === 'colors') && (bytesForOneValue === 4)) {
+      const colorValues = fileReader.readUint8Array(clusterCount * 4);
+      arraysDictionary[type] = [];
+      for (let j = 0; j < clusterCount; j++) {
+        arraysDictionary[type].push([]);
+        for (let k = 0; k < 4; k++) {
+          arraysDictionary[type][j].push(colorValues[j * 4 + k]);
+        }
+      }
+    } else if ((type === 'normals') && (bytesForOneValue === 12)) {
+      const normalValues = fileReader.readFloat32Array(clusterCount * 3);
+      arraysDictionary[type] = [];
+      for (let j = 0; j < clusterCount; j++) {
+        const newNormal = new THREE.Vector3(normalValues[j * 3], normalValues[j * 3 + 1], normalValues[j * 3 + 2]);
+        arraysDictionary[type].push(newNormal);
+      }
+    } else if (bytesForOneValue === 4) {
+      arraysDictionary[type] = fileReader.readFloat32Array(clusterCount);
+    } else {
+      throw Error('Well this is unexpected. bytesForOneValue is ' + bytesForOneValue);
+    }
+  }
+
+  const geometryAttributes = [];
+  while (true) {
+    const geometryType = fileReader.readUint32();
+    const geometryCount = fileReader.readUint32();
+    const attributeCount = fileReader.readUint32();
+    const byteCount = fileReader.readUint32();
+
+    const indexes = new FibonacciDecoder(asArrayBuffer, fileReader.location, byteCount);
+    fileReader.skip(byteCount);
+
+    geometryAttributes.push({'type': geometryType, 'indexes': indexes, 'count': geometryCount,
+    'attributeCount': attributeCount});
+
+    if (fileReader.location === asArrayBuffer.byteLength) {
+      break;
+    }
+  }
+
+  return {'magicBytes': magicBytes, 'formatVersion': formatVersion, 'optimizerVersion': optimizerVersion,
+  'arrayCount': arrayCount, 'arrays': arraysDictionary, 'geometries': geometryAttributes};
+}
+
+let nodeId: number;
+let treeIndex: number;
+let color: any;
+let x: number;
+let y: number;
+let z: number;
+let normal: THREE.Vector3;
+let deltaX: number;
+let deltaY: number;
+let deltaZ: number;
+let height: number;
+let radius: number;
+let radiusA: number;
+let radiusB: number;
+let angle: number;
+const center = new THREE.Vector3();
+const delta = new THREE.Vector3();
+
+const centerA = new THREE.Vector3();
+const centerB = new THREE.Vector3();
+
+function loadData(trueValuesArray: any, geometryInfo: any) {
+  nodeId =    geometryInfo.indexes.nextValue();
+  treeIndex = geometryInfo.indexes.nextValue();
+  color =     geometryInfo.indexes.nextValue();
+  const xIndex = geometryInfo.indexes.nextValue();
+  const yIndex = geometryInfo.indexes.nextValue();
+  const zIndex = geometryInfo.indexes.nextValue();
+
+  x =         trueValuesArray.x_values[xIndex];
+  y =         trueValuesArray.y_values[yIndex];
+  z =         trueValuesArray.z_values[zIndex];
+
+  center.set(x, y, z);
+
+  // Normal: All but spheres
+  if ([1, 3, 4].indexOf(geometryInfo.type) !== -1) {
```

Comment:
Could we use some sort of map so we can use readable strings? What is 1,3 and 4?

Reply:
Done